### DISCUSSION TITLE
Altersaddle Game Mode Mega Thread

### DISCUSSION THREAD

**Discussion by andrewdmaclean** [BattlesnakeOfficial/feedback#46] [General]
Recommendations made on the Legacy Battlesnake Slack
- No more lambas bread: Each food your snake eats is worth less and less over time.  The first one gives you 100 health, the second 95, third 90, etc.  Alternative: a different function (you get more if you wait longer?) to require strategic, rather than competitive, eating.
- Pay to turn: steering is expensive!  Snake health is only reduced if it chooses a move that is not the same as the last move.  When health is reduced , it is by an amount equal to the length of the snake.  Developers must weigh the expense of a u-turn or wriggly path, and the cost keeps going up!
- Thin ice: Normal rules on a large 19x19 board to start.  If a snake can successfully create a straight line with its body from one side of the board to the other [on an even-number-indexed row or column], then the board is cut along the successful snake body and the board is resized.  If the board is not equally divided, the larger section remains.  If they're equal, the remaining area that contains the successful snakes' head is retained.  The rest is removed from the game.  Any snake whose head is now out of bounds is eliminated; snakes whose heads are inbound have their tails stacked (as if they have eaten) - there should only be one of these anyway.  Some additional rules will need to be implemented for tiebreakers if two snakes cross the board in parallel at the same time. (edited) 
- This challenge might be more interesting with a bot opponent that starts at length board['width'] - 1 or something, so the threat of losing available play space is immediate.
- Countdown mode: The snakes' initial health is 500, but is not replenished when your snake eats.  Food adds 1 to your length and subtracts some amount from your health.  (50? 15?)
- variant rule: the initial health is 500 (or more), food doesn't replenish health, and each turn costs an amount of health equal to your snakes' length.
- Best before: Uneaten food becomes something else after a certain number of turns or (yikes!) a certain amount of time.  Maybe the become hazards, maybe they become impassible terrain, maybe they become bombs that... explode?
- Turbo food: When your snake consumes food, the next move it takes advances the snake head two (or three?) squares.  This has an impact on tail behavior (the tail no longer 'stays put' when the snake eats), plus some crazy / fascinating problems when trying to predict where your snake (and other snakes) might be.  Variant: rather than doubling the snake's chosen move, a random move could be added before/after the chosen move (spicy mode) - or perhaps the double/fast movement persists for a few turns.  Having the ability to tactically overtake or outrun another snake is quite the challenge.
- Snails & Slugs mode: A variant on pinned-tail/tron.  Your snake moves as normal/expected, but squares vacated by snakes are set to be hazard squares on the board (taking rules from royale mode for hazards effect, but appearing behind snake paths rather than from out edges of the board) (edited) 
- alternative rule: the hazards DO grow over time
- Ourobourous Mode: A variant on "eating a snake grows yours by its length":
- Your snake may safely collide with the tail of another different snake.
- If you do, your snake is now attached to that snake tail/body. 
- When attached, if you choose a move that matches that snake's move, your snake is advanced up the other snake body one square (towards its head).  Your snake length increases by 1.
- When attached, If you choose a move that is the reverse of that snake's move, your snake is eliminated.  Gross.
- Otherwise, your move is discarded and you're just here for the ride, your snake follows the other snake body.
- Head collisions work as before, except (presumably) you're going to win by having a longer snake.  There are interesting eating-collision order problems that might come up due to this strange arrangement.
- Slinky™ Snake: a 1v1 arena versus Slinky™ Snake.  Slinky™ Snake behaves like a normal snake, except you may overlap your snake with its body.  If you do, then Slinky™ Snake will extend in length for every turn that its body is 'blocked' by yours, adding new game rules:
- If Slinky™ Snake's length is double yours while your bodies overlap, your snake is removed from the game
- If Slinky™ Snake's length is doubled while your bodies overlap, Slinky™ is removed from the game
- When the body of Slinky™ Snake is no longer blocked, it returns to its pre-overlap size (snapping forward)
- Slinky™ Snake cannot eat while its body is extended
- Challenge idea: eating an enemy snake causes your snake to grow by their length
- squads with revives: if you pass over all squares your squadmate occupied, they are revived but your snakes' lengths are halved
- (variant: you must visit those squares in head-to-tail order)
- tractor pull: the longer your snake, the fewer ms allowed before timeout
- javelin mode: to win a head-to-head, your snake must be both longer and completely straight
- Another one: you start with several copies of your snake and have one kill all the others before dying. Could be an interesting one because since they're all your snake, you can coordinate to make kills easier.
- Pathfinding challenge: Board starts with exactly one food, but you only have exactly as much health as required to reach it, nothing more. When you eat the food, a new one appears and your health resets to the amount required to reach that new food. Snake devs will have to learn pathfinding algorithms to find the shortest path to the food.
- Tail-chasing challenge: Try to chase your tail in a loop, and your score is how long your snake's loop is when you start chasing. Or maybe how long you remain chasing, but that's limited by health

**Comment by altersaddle** [comment]
Since I can't edit my own fabulous list of ideas, I'll add more here:

Pumpkin Spice Challenge
- The game board is divided into four quadrants.  Food in each quadrant is assigned to: nutmeg, cinnamon, clove, ginger.
- There's one square down the middle of each axis that would not have a 'spice assignment'
- Variant 1: Your snake accumulates point(s) for every set of four spices it collects.
- Variant 2: The snake with the best balance of spices at the end of game has created the most delicious blend, and is awarded point(s).
- The primary goal of this challenge is to have the most points.  The secondary goal is to teach the world that pumpkin spice contains no pumpkin.
